docs/framework/interop/default-marshaling-for-strings.md

```diff
@@ -1,264 +1,280 @@
+
+```csharp
+public interface IStringWorker
+{
+    void PassString1(string s);
+    void PassString2([MarshalAs(UnmanagedType.BStr)] string s);
+    void PassString3([MarshalAs(UnmanagedType.LPStr)] string s);
+    void PassString4([MarshalAs(UnmanagedType.LPWStr)] string s);
+    void PassStringRef1(ref string s);
+    void PassStringRef2([MarshalAs(UnmanagedType.BStr)] ref string s);
+    void PassStringRef3([MarshalAs(UnmanagedType.LPStr)] ref string s);
+    void PassStringRef4([MarshalAs(UnmanagedType.LPWStr)] ref string s);
+}
+```
+
+```vb
+Public Interface IStringWorker
+    Sub PassString1(s As String)
+    Sub PassString2(<MarshalAs(UnmanagedType.BStr)> s As String)
+    Sub PassString3(<MarshalAs(UnmanagedType.LPStr)> s As String)
+    Sub PassString4(<MarshalAs(UnmanagedType.LPWStr)> s As String)
+    Sub PassStringRef1(ByRef s As String)
+    Sub PassStringRef2(<MarshalAs(UnmanagedType.BStr)> ByRef s As String)
+    Sub PassStringRef3(<MarshalAs(UnmanagedType.LPStr)> ByRef s As String)
+    Sub PassStringRef4(<MarshalAs(UnmanagedType.LPWStr)> ByRef s As String)
+End Interface
 ```
 
 The following example shows the corresponding interface described in a type library.
 
+```cpp
+interface IStringWorker : IDispatch
+{
+    HRESULT PassString1([in] BSTR s);
+    HRESULT PassString2([in] BSTR s);
+    HRESULT PassString3([in] LPStr s);
+    HRESULT PassString4([in] LPWStr s);
+    HRESULT PassStringRef1([in, out] BSTR *s);
+    HRESULT PassStringRef2([in, out] BSTR *s);
+    HRESULT PassStringRef3([in, out] LPStr *s);
+    HRESULT PassStringRef4([in, out] LPWStr *s);
+};
 ```
-[…]  
-interface IStringWorker : IDispatch {  
-HRESULT PassString1([in] BSTR s);  
-HRESULT PassString2([in] BSTR s);  
-HRESULT PassString3([in] LPStr s);  
-HRESULT PassString4([in] LPWStr s);  
-HRESULT PassStringRef1([in, out] BSTR *s);  
-HRESULT PassStringRef2([in, out] BSTR *s);  
-HRESULT PassStringRef3([in, out] LPStr *s);  
-HRESULT PassStringRef4([in, out] LPWStr *s);  
-);
+
+## Strings Used in Platform Invoke
+
+Platform invoke copies string arguments, converting from the .NET Framework format (Unicode) to the platform unmanaged format. Strings are immutable and are not copied back from unmanaged memory to managed memory when the call returns.
+
+The following table lists the marshaling options for strings when marshaled as a method argument of a platform invoke call. The <xref:System.Runtime.InteropServices.MarshalAsAttribute> attribute provides several <xref:System.Runtime.InteropServices.UnmanagedType> enumeration values to marshal strings.
+
+|Enumeration type|Description of unmanaged format|
+|----------------------|-------------------------------------|
+|`UnmanagedType.AnsiBStr`|A COM-style `BSTR` with a prefixed length and ANSI characters.|
+|`UnmanagedType.BStr`|A COM-style `BSTR` with a prefixed length and Unicode characters.|
+|`UnmanagedType.LPStr` (default)|A pointer to a null-terminated array of ANSI characters.|
```

Comment:
In [CharSet Enum](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.charset):

> Although the common language runtime default is `Auto`, languages may override this default. For example, by default C# marks all methods and types as `Ansi`.

In [DllImportAttribute.CharSet Field](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.dllimportattribute.charset):

> The default enumeration member for C# and Visual Basic is `CharSet.Ansi` and the default enumeration member for C++ is `CharSet.None`, which is equivalent to `CharSet.Ansi`.

`Auto` is the CLR's default but `Ansi` is the languages' default, which actually takes effect 😕. What are your thoughts?